### ISSUE TITLE
Linux Version

### ISSUE THREAD

**Issue by ethangamma24** [bkacjios/m-overlay#1]
How hard would it be to get this working in Linux? I've been able to launch the overlay using Wine, but it's looking for Dolphin.exe, not the 'dolphin-emu' (the process name in Linux). Is it as simple as changing the name of the process to look for in the code?

**Comment by bkacjios** [comment]
It's possible, but running the windows version in wine won't work right, unless you ran dolphin under wine as well.

I would have to create a new module for reading memory in linux since currently I use a lot of windows only API calls. It shouldn't be too hard, seeing how what I do is very similar to what Dolphin-memory-engine uses, and they support linux, so.. I could try to get it working.

**Comment by bkacjios** [comment]
Well, I got it working pretty easily.

![Linux test](https://i.imgur.com/1tYI7Gx.png)

You have to run it as root for it to work though, which is a little annoying.

**Comment by bkacjios** [comment]
Alright, mind giving this a try?

https://bitassemble.com/m-overlay/m-overlay-x64-linux (Slippi).love

Make sure you have love2d 11.2 installed for linux. https://love2d.org/

**Comment by ethangamma24** [comment]
Both are just showing up as a blue screen.
![Screenshot_20190711_174124](https://user-images.githubusercontent.com/16707967/61087248-1d9a0100-a403-11e9-98cf-a3c07e3d70a6.png)

**Comment by ethangamma24** [comment]
I'm using Manjaro if that makes a difference

**Comment by bkacjios** [comment]
If you run it in a terminal, are there any errors?

**Comment by bkacjios** [comment]
Alright, I made a few changes that may fix your issue. I re-uploaded the builds. If it errors, the blue screen _should_ show an error log

[Linux x64](https://bitassemble.com/m-overlay/m-overlay-x64-linux.love)
[Linux x64 (Slippi)](https://bitassemble.com/m-overlay/m-overlay-x64-linux%20(Slippi).love)

**Comment by ethangamma24** [comment]
Sorry I'm just now getting back to you! I forgot. Here's a screenshot of the error:
![Screenshot_20190716_200318](https://user-images.githubusercontent.com/16707967/61337805-d649b100-a804-11e9-800e-b4800b0f939d.png)

**Comment by bkacjios** [comment]
Hmmm, that's really strange.. That honestly shouldn't be happening, but I made a change that should work around you not having ssize_t defined for whatever reason..

[Linux x64](https://bitassemble.com/m-overlay/m-overlay-x64-linux.love)
[Linux x64 (Slippi)](https://bitassemble.com/m-overlay/m-overlay-x64-linux%20(Slippi).love)

**Comment by bkacjios** [comment]
Also worth noting, I found a command that will allow love to access memory of other running applications, so you no longer have to run it as root.

```
sudo setcap cap_sys_ptrace=eip /usr/bin/love
```

After that, you can run the .love file like it were any application.

**Comment by ethangamma24** [comment]
Don't have to run as root anymore, but I'm still getting the same error D:

**Comment by bkacjios** [comment]
Redownload the files then, because you still are trying to run the old version.

**Comment by ethangamma24** [comment]
I'm not, this is the new download. 
![image](https://user-images.githubusercontent.com/16707967/61572869-7f491380-aa72-11e9-9c5c-fab33a89115c.png)

**Comment by bkacjios** [comment]
Well, that error isn't really possible..

![Source](https://i.imgur.com/ozMC8Yq.png)

I changed ssize_t to int

**Comment by ethangamma24** [comment]
Awesome! Okay, the overlay shows now, but even when I start Melee, it still says "Waiting for Dolphin".

**Comment by bkacjios** [comment]
Even if you run it as root?

**Comment by ethangamma24** [comment]
Yeah even then

**Comment by bkacjios** [comment]
![Working in Manjaro?](https://i.imgur.com/matGRNw.png)

So, I installed a VM of Manjaro just to test things out.. It seems to hook perfectly fine for me, so I'm really not sure why you're having all these issues.

I can't test it fully, since I can't run the game in the VM, but I think it would be fine?

What version of Dolphin are you running? I've tested it on both Vanilla dolphin and Slippi FM r18
